### PR TITLE
Update xClusterDisk from disk number to guid

### PR DESCRIPTION
**Pull request by claudiospizzi**
Hi

I've tried to use the xClusterDisk resource on a target system, where the disk numbers are not matching and are reused multiple times, because the cluster has attached local dedicated SAN disks as well as shares clustered SAN disks. As a conclusion, the disk number cannot be used as a identifier for the cluster disk - it's just not working in every case.

Therefore, I've reworked the resource to now use the disks id (aka guid, signature) for identification. Tests have been reworked too.

Because we are changing a key property in the MOF, this will be a breaking change.

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/xfailovercluster/26)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 
 ### Unreleased
 
+* BREAKING CHANGE: Update xClusterDisk from disk number to guid (aka id, signature)
```
Comment (TravisEz13) [inline README.md]: Add a description of why

File: DSCResources/MSFT_xClusterDisk/MSFT_xClusterDisk.schema.mof [hunk 2]
```diff
@@ -2,7 +2,7 @@
 [ClassVersion("1.0.0.0"), FriendlyName("xClusterDisk")]
 class MSFT_xClusterDisk : OMI_BaseResource
 {
-    [Key] String Number;
     [Write,ValueMap{"Present", "Absent"},Values{"Present", "Absent"}] string Ensure;
+    [Key] String Guid;
```
Comment (TravisEz13) [inline DSCResources/MSFT_xClusterDisk/MSFT_xClusterDisk.schema.mof]: Please move the key to the first.  
A description would be nice

File: Tests/MSFT_xClusterDisk.Tests.ps1 [hunk 3]
```diff
@@ -12,191 +7,248 @@ if (!$PSScriptRoot)
 $RootPath   = (Resolve-Path -Path "$PSScriptRoot\..").Path
 $ModuleName = 'MSFT_xClusterDisk'
 
-Add-WindowsFeature -Name RSAT-Clustering-PowerShell -ErrorAction SilentlyContinue    
+try
```
Comment (TravisEz13) [inline Tests/MSFT_xClusterDisk.Tests.ps1]: Don't use catch to detect the OS Product Type.  Query it.  Code for WMF5.0 and lower where DSC is supported:
```PowerShell
if((Get-CimInstance 'Win32_OperatingSystem').ProductType -ne 1)
{
    # For server operating system
    Add-WindowsFeature -Name RSAT-Clustering-PowerShell -ErrorAction Stop
}
else
{
    # For client operating system
    Enable-WindowsOptionalFeature -Online -FeatureName 'RSATClient-Features-Clustering' -ErrorAction Stop
}
```
On WMF 5.1, `Get-ComputerInfo` will have more information about nano, but I don't think it is needed here.

**Comment by claudiospizzi** [comment]
@TravisEz13 I've updated the pull request according to your suggestions.

**Comment by johlju** [comment]
I think it would be better to actually keep the Disk number as the key, and add Guid as an optional parameter. It would then correlate with xDisk in xStorage module which uses only disk number.
If `Guid` is used by the user, then `Number` parameter is ignore (should be stated in README.md), and disks are evaluated using the Guid only.
With this approach it is not a breaking change, and both scenarios work.

**Comment by johlju** [comment]
I think this proposed change should not be merged according to new issue #39. Unless the Guid is a better solution for Cluster Disks.

**Comment by johlju** [comment]
@claudiospizzi Please let me know if you are still working on this PR? I will label this as abandoned in the mean time.

**Comment by johlju** [comment]
@bgelens and @RichardSiddaway what is your thought regarding this PR and the issue raised in #39? In regards to Guid is not optimal for choosing disk, but rather UniqueID is better, and also line up with the resource in xStorage.

**Comment by claudiospizzi** [comment]
@johlju Quite an old PR, completely lost track about that. Do you have the view on the resource, does this PR still makes sense? I think we should decide to put effort in it and review, fix and merge it - or close it.

**Comment by johlju** [comment]
@claudiospizzi I think something along the lines is needed, but I think we need to look at the work that has been done in StorageDsc for the [Disk resource](https://github.com/PowerShell/StorageDsc/wiki/Disk). I think we need to do something similar, not have one or the other, but having both (or three types in the case of Disk). This way it would correlate when using the Disk resource together with this resource. Same type ID can be used throughout. Make sense?

If this PR could be refactored to that that would be great, but if you feel it would be easier to start from the beginning (by you or someone else) then I'm okay with this PR get closed.

**Comment by johlju** [comment]
Closing this PR due to inactivity. If there are someone who wants to work on this issue or PR please open a new PR.

**Pull request closed without merging**